third_party/move/move-vm/runtime/src/storage/implementations/unsync_module_storage.rs

```diff
@@ -148,28 +150,22 @@ impl<'e, B: ModuleBytesStorage> UnsyncModuleStorage<'e, B> {
 
         if !self.is_module_cached(address, module_name) {
             let bytes = self.get_existing_module_bytes(address, module_name)?;
-            let compiled_module = CompiledModule::deserialize_with_config(
-                &bytes,
-                &self.runtime_environment.vm_config().deserializer_config,
-            )
-            .map_err(|err| {
-                let msg = format!("Deserialization error: {:?}", err);
-                PartialVMError::new(StatusCode::CODE_DESERIALIZATION_ERROR)
-                    .with_message(msg)
-                    .finish(Location::Module(ModuleId::new(
-                        *address,
-                        module_name.to_owned(),
-                    )))
-            })?;
+            let (module, module_size, module_hash) = self
+                .runtime_environment
+                .deserialize_into_compiled_module(&bytes)?;
+            self.runtime_environment
```

Comment:
is this a new check?

Reply:
Paranoid? Yes, loader V1 had it so I added ti here as well to mimic the same behaviour.

Reply:
I followed it in a number of places, but here can't find where V1 was calling paranoid check? it wasn't inside deserialize_with_config, right? did we remove it before, or was it happening outside of initialize_module_storage_entry and now it's moved inside?

Reply:
It was done after the code was loaded from data store here: https://github.com/aptos-labs/aptos-core/blob/78c815b8a7b8a42a861e446b624d7587b304dc86/third_party/move/move-vm/runtime/src/loader/mod.rs#L886